### PR TITLE
Add Hindi experiment Related Content - Var B

### LINKED ISSUE #9870: Hindi experiment: - Var B - Related content
**Is your feature request related to a problem? Please describe.**
Overview

This will show 4 promos in the ‘Related content’ component replacing the editorially chosen content

The title will stay the same as the title on the Recommendations component

Up to 4 promos will be shown in order based on the recommendations endpoint

Dates will not be shown in the promos

Optimizley view and click events will need to be added to the whole block (this is currently implemented in ATI)

UX
Experiment Designs: https://zpl.io/aXyyrE8

Live CPS Related Content designs: https://app.zeplin.io/project/5baa564c05c2eba0d2b89568/screen/613a13351714592686b37888

A11y
This should be the same as what’s live for current related content:
Live CPS Related content (labelled as 'More on this story' on the design): https://app.zeplin.io/project/5baa564c05c2eba0d2b89568/screen/6125660245e314a3399ef4d3

Test
Use the whitelist in Optimizley to show variation on Test

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Checklist**
- [ ] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**

### PR DESCRIPTION
**Pull request by amoore108**
Resolves [#9870](https://github.com/bbc/simorgh/issues/9870)

**Overall change:**
- Adds Optimizely event tracking to the Related Content component
- Swaps out the Related Content data with Recommended Stories data if the user is bucketed into `variation_2` of the Hindi experiment, otherwise it uses the original Related Content data

**Code changes:**
- Adds the `withOptimizelyProvider` HOC to the `CpsRelatedContent` component. This is needed as this component is used on pages outside of the StoryPage that don't have the `withOptimizelyProvider` wrapped around them
- Adds Optimizely context to the event tracking data object if in the variation

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project
- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/pages/StoryPage/StoryPage.jsx [hunk 1]
```diff
@@ -393,7 +393,9 @@ const StoryPage = ({ pageData, mostReadEndpointOverride }) => {
 
           <CpsRelatedContent
```
Comment (MuhammadAHussain) [inline src/app/pages/StoryPage/StoryPage.jsx:395]: Would be worth having a test in story page to assert recommendations isn't rendered when variation is `control` for example.

**Review by MuhammadAHussain — APPROVED** [review]
Left a comment. LGTM 👍 otherwise.

File: src/app/containers/CpsRelatedContent/index.test.jsx [hunk 2]
```diff
@@ -143,9 +128,42 @@ describe('CpsRelatedContent', () => {
     expect(getByText('February 2009', { exact: false })).not.toBeNull();
   });
   it('should render a default title if translations are not available', () => {
-    renderRelatedContentNoTitle();
+    renderRelatedContent({ pageType: STORY_PAGE, service: 'news' });
     expect(screen.getByText(`Related content`)).toBeTruthy();
   });
+
+  it('should render a translated title for Hindi Recommendations Related Content', async () => {
```
Comment (MuhammadAHussain) [inline src/app/containers/CpsRelatedContent/index.test.jsx]: Forgot to mention, probably a good idea to wrap the test created for the experiment in describe blocks (I followed the pattern below) like:
```
  describe('Optimizely Experiments', () => {
    describe('003_hindi_experiment_feature', () => {
      describe('variation_b', () => {});
    });
  });
```
Makes it easier to teardown in my opinion.

**Review by DarioR01 — APPROVED** [review]
LGTM

**Pull request merged**